### DISCUSSION TITLE
RFC: Improve or remove setting an image on taxons

### DISCUSSION THREAD

**Discussion by kennyadsl** [solidusio/solidus#4547] [New Features or Ideas]
**Is your feature request related to a problem? Please describe.**

At the moment we can set a taxon icon but we cannot set multiple images, sort them and there's no way to remove an image already set (see #3349).

<img width="1032" alt="Schermata 2019-10-07 alle 15 12 16" src="https://user-images.githubusercontent.com/167946/66314862-fc1eb780-e914-11e9-88fc-a6046b1c3fe7.png">

As is, this feature does not give a lot of value to Solidus. Almost any real store implementation of this requirement would need some change to the code, and considering how easy it is to setup a new image upload for a model on Rails, maybe, it would be better to remove this feature (after a deprecation period) and create a wiki/guide page that explains in detail how to do that in your store, since this is a recurring need.

**Describe alternatives you've considered**

Alternatively, we can improve this section in the core adding the ability to create multiple images, remove and sort them, with a fancy interface as we are doing in the Images tab of products. Still, there's the risk that this would need some heavy modification from developers, but at least it would be an acceptable default.

<img width="1172" alt="Schermata 2019-10-07 alle 15 11 57" src="https://user-images.githubusercontent.com/167946/66314891-08a31000-e915-11e9-9370-614850499950.png">

**Feedback**

Some feedback from how this feature has been used in real stores is appreciated to take the right path here. 🙏

**Comment by ericsaupe** [comment]
I'd be for updating the interface in core. The better the support for great images around the app is the better the stores look.

**Comment by seand7565** [comment]
If we improve on how we deal with Spree::Asset, we can use them to store taxon icons via a has_many relationship. #3609
